File: Random Marathon 64/C 10769.py
'''
import sys
input = sys.stdin.readline

arr = input().rstrip()

prev = ""
happy = 0
sad = 0

for char in arr:

    if char == ":":
        prev = ":"

    elif prev == ":":
        if char == "-":
            prev = "-"
        else:
            prev = ""

    elif prev == "-":
        if char == ")":
            happy += 1
            prev = ""
        elif char == "(":
            sad += 1
            prev = ""
        else:
            prev = ""

    else:
        continue

if happy == sad == 0:
    print("none")
elif happy == sad:
    print("unsure")
elif happy > sad:
    print("happy")
elif happy < sad:
    print("sad")

# 이걸 c언어로 할 수 있겠지? 도전이다.
'''

import sys
input = sys.stdin.readline

arr = input().rstrip()

happy = 0
sad = 0

for i in range(len(arr) - 2):
    if arr[i] == ':' and arr[i + 1] == '-':
        if arr[i + 2] == '(':
            sad += 1
        elif arr[i + 2] == ')':
            happy += 1

if happy == sad == 0:
    print("none")
elif happy == sad:
    print("unsure")
elif happy > sad:
    print("happy")
elif happy < sad:
    print("sad")

File: Random Marathon/week 64/C 10769.c
/*
#include <stdio.h>
#include <string.h>

int main(void) {

    char arr[256];
    fgets(arr, sizeof(arr), stdin);

    char prev = '\0';
    int happy = 0, sad = 0;

    size_t len = strlen(arr);
    for (size_t i = 0; i < len; i++) {

        if (arr[i] == ':') {
            prev = ':';
        }
        else if  (prev == ':') {
            if (arr[i] == '-') {
                prev = '-';
            }
            else {
                prev = '\0';
            }
        }
        else if (prev == '-') {
            if (arr[i] == ')'){
                happy += 1;
                prev = '\0';
            }
            else if (arr[i] == '(') {
                sad += 1;
                prev = '\0';
            }
            else {
                prev = '\0';
            }
        }
        else {
            continue;
        }
    }

    if (happy == 0 && sad == 0) printf("none");
    else if (happy == sad) printf("unsure");
    else if (happy > sad) printf("happy");
    else if (happy < sad) printf("sad");

    return 0;
}
*/

#include <stdio.h>
#include <string.h>

int main(void) {

    char arr[256];
    fgets(arr, sizeof(arr), stdin);

    int happy = 0, sad = 0;

    size_t len = strlen(arr);
    for (size_t i = 0; i + 2 < len; i++) {
        if (arr[i] == ':' && arr[i + 1] == '-') {
            if (arr[i + 2] == ')') happy++;
            else if (arr[i + 2] == '(') sad++;
        }
    }

    if (happy == 0 && sad == 0) printf("none");
    else if (happy == sad) printf("unsure");
    else if (happy > sad) printf("happy");
    else if (happy < sad) printf("sad");

    return 0;
}


